In [1]:
import os
import pandas as pd
from pathlib import Path
from collections import defaultdict
import numpy as np
import matplotlib.pyplot as plt
import datetime

In [2]:
away_team_score = pd.read_csv('../Output/away_team_score.csv')
away_team = pd.read_csv('../Output/away_team.csv')
event = pd.read_csv('../Output/event.csv')
home_team_score = pd.read_csv('../Output/home_team_score.csv')
home_team = pd.read_csv('../Output/home_team.csv')
raw_odds_parquet = pd.read_csv('../Output/raw_odds_parquet.csv')
raw_point_by_point_parquet = pd.read_csv('../Output/raw_point_by_point_parquet.csv')
raw_statistics_parquet = pd.read_csv('../Output/raw_statistics_parquet.csv')
raw_tennis_power_parquet = pd.read_csv('../Output/raw_tennis_power_parquet.csv')
raw_votes_parquet = pd.read_csv('../Output/raw_votes_parquet.csv')
round = pd.read_csv('../Output/round.csv')
season = pd.read_csv('../Output/season.csv')
time = pd.read_csv('../Output/time.csv')
tournament = pd.read_csv('../Output/tournament.csv')
venue = pd.read_csv('../Output/venue.csv')

how double faults changed from first set to last set:

In [3]:
double_faults = raw_statistics_parquet.loc[(raw_statistics_parquet['statistic_name']=='double_faults')].copy()
double_faults['sum_double_faults'] = double_faults['home_stat'].astype(int)+double_faults['away_stat'].astype(int)
double_faults_1 = double_faults.groupby('period')['sum_double_faults'].sum().reset_index()
double_faults_1['sum_double_faults'] = double_faults_1['sum_double_faults']/double_faults.loc[double_faults['period']=='ALL','sum_double_faults'].sum()
double_faults_1.drop(index=3,inplace=True)
double_faults_1['sum_double_faults'] = (double_faults_1['sum_double_faults']*100).round(3)
double_faults_1.rename(columns={'sum_double_faults':'percent_of_double_faults'})

period  percent_of_double_faults
0    1ST                    46.106
1    2ND                    42.236
2    3RD                    10.935

This shows that nearly 46% of double_faults occur in the first set, 42% in the second set, and only 11% in the third set. Although players become tired as the game gets longer and the likelihood of making more mistakes increases, it seems that this sensitivity of the game, which increases in the final sets, has a greater impact on the number of mistakes.